scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -1063,12 +1063,18 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
    * Save this SCollection as a Pub/Sub topic.
    * @group output
    */
-  def saveAsPubsub(topic: String): Future[Tap[T]] = {
+  def saveAsPubsub(topic: String,
```

Comment:
Same reasoning with the read counterpart, how about a variation that saves a `SCollection[(T, Map[String, String])]` by encoding `T` as message and `Map` as attributes?